sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c

```diff
@@ -394,7 +394,7 @@ static int _fsm_rcvd_pkt(gnrc_tcp_tcb_t *tcb, gnrc_pktsnip_t *in_pkt)
 {
     gnrc_pktsnip_t *out_pkt = NULL;  /* Outgoing packet */
     uint16_t seq_con = 0;            /* Sequence number consumption of outgoing packet */
-    gnrc_pktsnip_t *snp = NULL;      /* Temporary packet snip */
+    gnrc_pktsnip_t *snp;             /* Temporary packet snip */
```

Comment:
```suggestion
```

Reply:
Actually here I would prefer a re-ordering of the declarations to prevent memory padding on the stack.

```C
    gnrc_pktsnip_t *out_pkt = NULL;  /* Outgoing packet */
    gnrc_pktsnip_t *snp;             /* Temporary packet snip */

    gnrc_tcp_tcb_t *lst = NULL;      /* Temporary pointer to TCB */
    uint32_t seg_seq = 0;            /* Sequence number of the incoming packet*/
    uint32_t seg_ack = 0;            /* Acknowledgment number of the incoming packet */
    uint32_t seg_wnd = 0;            /* Receive window of the incoming packet */
    uint16_t ctl = 0;                /* Control bits of the incoming packet */
    uint16_t seq_con = 0;            /* Sequence number consumption of outgoing packet */
```

As this is an unrelated change however, and should go into a separate PR. As for this PR I would prefer to keep this line as is, if this is okay for you.

Reply:
With this reasoning its fine by me, but then I would like you to initialize snp with NULL

Reply:
Then `cppcheck` will complain, that there is a redundant assignment ;-).

Reply:
Its hard for me to tolerate a non-initialized pointer, how ever I'll hope cppcheck will warns if somebody tries to use it before its initialized. I would prefer to disable the check with /* cppcheck-suppress redundantAssignment */, but it is not manditory to pass this PR. Its up to you.

Regarding the stack reordering: I'll would like to see if there are any real size changes (within the according PR of course), I am pretty sure this is one of the easiest optimizations a compiler can do, so would be suprised if there are notable differences.

Reply:
> Its hard for me to tolerate a non-initialized pointer, how ever I'll hope cppcheck will warns if somebody tries to use it before its initialized. I would prefer to disable the check with /* cppcheck-suppress redundantAssignment */, but it is not manditory to pass this PR. Its up to you.

Not only `cppcheck` is warning you about that but most static analyzers including the GCC-buildin one. I don't like suppressing a true positive with a static analyzer. Redundant assigments mean more instructions, mean larger `.text` segment, mean more ROM usage, so I think the warning is justified.

> Regarding the stack reordering: I'll would like to see if there are any real size changes (within the according PR of course), I am pretty sure this is one of the easiest optimizations a compiler can do, so I would be suprised if there are notable differences.

One would think so, yes. But I seem to remember that there were issues with some of our compilers.